pkg/tbtc/chain.go

```diff
@@ -497,4 +451,6 @@ type Chain interface {
 	DistributedKeyGenerationChain
 	BridgeChain
 	WalletCoordinatorChain
+	ValidateDepositSweepProposalChain
+	ValidateRedemptionProposalChain
```

Comment:
So far, the structure of interfaces in the `chain.go` file reflected the contract structure we have on the chain. I think we should stick to that approach and avoid creating interfaces like `ValidateDepositSweepProposalChain`. If we go that way, we would soon have problems with defining the right boundaries between those interfaces. For example, `ValidateDepositSweepProposalChain` declares `GetPendingRedemptionRequest`. What if I need this function for a purpose other than proposal validation?

Reply:
> For example, `ValidateDepositSweepProposalChain` declares `GetPendingRedemptionRequest`. What if I need this function for a purpose other than proposal validation?

This is the case for #3649 currently in progress.

I do agree with this comment. I see it was done to avoid the duplication here:

https://github.com/keep-network/keep-core/blob/7eeea1f9e3a5aae0463244731e87cf5adf57b294/pkg/coordinator/chain.go#L112-L113

but I think having the duplication of function declarations is a better choice in this particular case. I am not a huge fan of importing types from `pkg/tbtc` as you already know from the other comment but importing interfaces makes it look even worse for me. 😉

Reply:
Alternative approach: 70bd4497a02fd50ade14bf471b78d0365e990055

Reply:
This approach is fine for me but I'd like to hear from @lukasz-zimnoch if it does not break anything for the automation part.

Reply:
Looks good!